api/src/main/java/marquez/api/OpenLineageResource.java

```diff
@@ -87,6 +89,96 @@ public void create(@Valid @NotNull BaseEvent event, @Suspended final AsyncRespon
     }
   }
 
+  private UUID runUuidFromEvent(LineageEvent.Run run) {
+    UUID runUuid;
+    try {
+      runUuid = UUID.fromString(run.getRunId());
+    } catch (Exception e) {
+      runUuid = UUID.nameUUIDFromBytes(run.getRunId().getBytes(StandardCharsets.UTF_8));
+    }
+    return runUuid;
+  }
+
+  private void indexEvent(@Valid @NotNull LineageEvent event) {
+    if (this.elasticsearchClient != null) {
+      UUID runUuid = runUuidFromEvent(event.getRun());
+      log.info("Indexing event {}", event);
+
+      if (event.getInputs() != null) {
+        indexDatasets(event.getInputs(), runUuid, event);
+      }
+      if (event.getOutputs() != null) {
+        indexDatasets(event.getOutputs(), runUuid, event);
+      }
+      indexJob(runUuid, event);
+    }
+  }
+
+  private Map<String, Object> buildJobIndexRequest(UUID runUuid, LineageEvent event) {
```

Comment:
How about creating separate class for this?

Reply:
Yeah, there is some work to be done reorganizing this code and also creating an independent service to manage the queries as well.